src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs

```diff
@@ -489,22 +489,26 @@ public override void ExecuteCmdlet()
 
                     AzureVmRecoveryPoint rp = (AzureVmRecoveryPoint)RecoveryPoint;
 
+                    string targetZone = TargetZoneNumber.ToString();
+
+                    if (TargetZoneNumber == 0)
+                    {
+                        targetZone = "NoZone";
+                    }
+
+                    WriteDebug("Target Zone = " + targetZone);
+
                     // eliminate LRS/GRS
                     if (storageType == AzureRmRecoveryServicesBackupStorageRedundancyType.ZoneRedundant.ToString() ||                     
                         (storageType == AzureRmRecoveryServicesBackupStorageRedundancyType.GeoRedundant.ToString() && crrEnabled))
                     {
                         // eliminate Archive tier RPs. Snapshot RPs are supported for RPCv2/Enhanced policy
                         // service would throw the appropriate error for Standard policy
-                        if (rp.RecoveryPointTier != 0 && rp.RecoveryPointTier != RecoveryPointTier.VaultArchive) 
+                        if (rp.RecoveryPointTier != 0 
+                            && rp.RecoveryPointTier != RecoveryPointTier.VaultArchive 
+                            && rp.RecoveryPointTier != RecoveryPointTier.SnapshotAndVaultArchive
+                            && rp.RecoveryPointTier != RecoveryPointTier.Snapshot) 
```

Comment:
are we allowing restore from SnapshotAndVaultStandard RP ? If yes, do we not have the 4HRs restriction on it now ?